<launch>

    <node pkg="remote_control" type="remote_control_node" name="remote_control" output="screen">
        <param name="active" value="true"/>

        <param name="frequency" value="50"/>
        <param name="MAX_linear_speed" value="0.6"/>    <!-- [m/s] -->
        <param name="MAX_angular_speed" value="1.2"/>   <!-- [rad/s] -->
        <param name="init_arm_x" value="128.0"/>        <!-- [mm] -->
        <param name="init_arm_y" value="17.0"/>         <!-- [mm] -->
        <param name="init_arm_z" value="10.0"/>         <!-- [mm] -->
        <param name="arm_MAX_XYspeed" value="10.0"/>    <!-- [mm/s] -->
        <param name="arm_MAX_XYspeed" value="5.0"/>     <!-- [mm/s] -->
  
        <param name="joy_topic" value="/nlink1/ds5_joy"/>   <!-- Sub [sensor_msgs::Joy] -->
        <param name="twist_topic" value="/cmd_vel"/>        <!-- Pub [geometry_msgs::Twist] -->
        <param name="point_topic" value="/arm_goal"/>       <!-- Pub [geometry_msgs::Point] -->
        <param name="suck_topic" value="/suck"/>            <!-- Pub [std_msgs::Bool] -->
    </node>

    <group ns="nlink1">
        <node pkg="nlink_parser" type="linktrack" name="linktrack0" output="screen">
            <!-- Testing on laptop -->
            <!-- <param name="port_name" value="/dev/ttyUSB1" />  -->
            <!-- Using on Raspberry Pi 4 -->
            <param name="port_name" value="/dev/USB0"/>
            <param name="baud_rate" value="921600"/>
        </node>
        <node pkg="remote_control" type="joy_deserialization" name="joy_deserialization" output="screen">
            <remap from="node_frame" to="nlink_linktrack_nodeframe0"/>
        </node>
    </group>

</launch>